addons/base_vat/models/res_partner.py

```diff
@@ -292,6 +292,27 @@ def check_vat_al(self, vat):
             return True
         return False
 
+    __check_tin1_ro_natural_persons = re.compile(r'[1-9]\d{2}(0[1-9]|1[0-2])(0[1-9]|[12]\d|3[01])\d{6}')
```

Comment:
If I'm not wrong, the last part is wrong as well. We're too permissive in the regex according to the documentation: 
![image](https://github.com/odoo/odoo/assets/6582885/6a2ca92c-3964-406c-8c9b-a66c1e25ddfd)
Btw if we want to verify strictly the TIN, we probably have to check that we don't have more days than what the month has :nerd_face:  (don't do it)

Reply:
For this i thought that adding the last part would be a bit messy in the regex that is already a bit awful ahah, but i can totally add it :smile: